src/egraph.rs

```diff
@@ -519,88 +576,51 @@ impl<L: Language, N: Analysis<L>> EGraph<L, N> {
     }
 
     /// Given two patterns and a substitution, add the patterns
-    /// and mark them for unioning.
-    /// The unions are performed when [`rebuild`](EGraph::rebuild) is called.
+    /// and union them.
+    ///
     /// When explanations are enabled [`with_explanations_enabled`](Runner::with_explanations_enabled), use
     /// this function instead of [`union`](EGraph::union).
     ///
-    /// The returned `bool` indicates whether a union is necessary,
-    /// and returned Id represents the eclass of the left pattern.
+    /// Returns the id of the new eclass, along with
+    /// a `bool` indicating whether a union occured.
     pub fn union_instantiations(
         &mut self,
         from_pat: &PatternAst<L>,
         to_pat: &PatternAst<L>,
         subst: &Subst,
         rule_name: impl Into<Symbol>,
     ) -> (Id, bool) {
-        let id1 = self.add_instantiation(from_pat, subst);
+        let id1 = self.add_instantiation_internal(from_pat, subst);
```

Comment:
"Internal" functions now return uncanonicalized ids so we can track which enode explanations use.